components/tabs/tab-mixin.js

```diff
@@ -0,0 +1,161 @@
+import '../colors/colors.js';
+import { css, html, unsafeCSS } from 'lit';
+import { classMap } from 'lit/directives/class-map.js';
+import { getFocusPseudoClass } from '../../helpers/focus.js';
+import ResizeObserver from 'resize-observer-polyfill/dist/ResizeObserver.es.js';
+
+const keyCodes = {
+	ENTER: 13,
+	SPACE: 32
+};
+
+export const TabMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			selected: { type: Boolean, reflect: true },
+			tabIndex: { type: Number },
+		};
+	};
+
+	static styles = css`
+        :host {
+            box-sizing: border-box;
+            display: inline-block;
+            max-width: 200px;
+            outline: none;
+            position: relative;
+            vertical-align: middle;
+        }
+        .d2l-tab-text {
+            margin: 0.5rem;
+            overflow: hidden;
+            padding: 0.1rem;
+            text-overflow: ellipsis;
+            white-space: nowrap;
+        }
+        :host(:first-child) .d2l-tab-text {
+            margin-left: 0;
```

Comment:
I think we'll need to handle the RTL case. Our preferred way is with CSS logical properties. In this case we can have the browser automagically handle it with:
```css
margin-inline-start: 0;
```
(previously, we would have used `RtlMixin`, which shouldn't be needed if we use the logical properties)